test/NuGet.Clients.Tests/NuGet.CommandLine.Test/NuGetSourcesCommandTest.cs

```diff
@@ -42,6 +42,214 @@ public void SourcesCommandTest_AddSource()
+                var sourceItem = packageSourcesSection?.GetFirstItemWithAttribute<SourceItem>("key", "test_source");
+                Assert.Equal("http://test_source", sourceItem.GetValueAsPath());
+                Assert.True(result.Output.Contains("WARNING: You are running the 'update source' operation with an 'http' source, 'http://test_source'. Support for 'http' sources will be removed in a future version."));
+            }
+        }
+
+        [Fact]
+        public void SourcesCommandTest_EnableSource_WarnWhenUsingHttp()
+        {
+            // Arrange
+            var nugetexe = Util.GetNuGetExePath();
+
+            using (var configFileDirectory = TestDirectory.Create())
+            {
+                var configFileName = "nuget.config";
+                var configFilePath = Path.Combine(configFileDirectory, configFileName);
+
+                Util.CreateFile(configFileDirectory, configFileName,
+                    @"<?xml version=""1.0"" encoding=""utf-8""?>
+<configuration>
+  <packageSources>
+    <add key=""test_source"" value=""http://test_source"" />
+  </packageSources>
+  <disabledPackageSources>
+    <add key=""test_source"" value=""true"" />
+    <add key=""Microsoft and .NET"" value=""true"" />
+  </disabledPackageSources>
+</configuration>");
+
+                var args = new string[] {
+                    "sources",
+                    "Enable",
+                    "-Name",
+                    "test_source",
+                    "-ConfigFile",
+                    configFilePath
+                };
+
+                // Act
+                var result = CommandRunner.Run(
+                    nugetexe,
+                    Directory.GetCurrentDirectory(),
+                    string.Join(" ", args),
+                    true);
+
+                // Assert
+                Util.VerifyResultSuccess(result);
+
+                var settings = Configuration.Settings.LoadDefaultSettings(
+                    configFileDirectory,
+                    configFileName,
+                    null);
+
+                var packageSourceProvider = new Configuration.PackageSourceProvider(settings);
+                var sources = packageSourceProvider.LoadPackageSources().ToList();
+
+                var testSources = sources.Where(s => s.Name == "test_source");
+                Assert.Single(testSources);
+                var source = testSources.Single();
+
+                Assert.Equal("test_source", source.Name);
+                Assert.Equal("http://test_source", source.Source);
+                Assert.True(source.IsEnabled, "Source is not enabled");
+                Assert.True(result.Output.Contains("WARNING: You are running the 'enable source' operation with an 'http' source, 'http://test_source'. Support for 'http' sources will be removed in a future version."));
+            }
+        }
+
+        [Fact]
+        public void SourcesCommandTest_DisableSource_WarnWhenUsingHttp()
```

Comment:
This is surprising. If it's disabled, why do we still warn?

Reply:
Oh this is just for sources command, so would that be here?

```
Registered Sources:
  1.  nuget.org [Disabled]
      http://api.nuget.org/v3/index.json
            <!--insert HTTP warning here ?? -->
  2.  dotnet6 [Enabled]
      https://aka.ms/dotnet6/nuget/index.json
```

Reply:
Yess it will look like that